CoreUtilities/src/au/gov/asd/tac/constellation/utilities/svg/SVGObject.java

```diff
@@ -0,0 +1,517 @@
+        return child != null ? new SVGObject(child) : null;
+    }
+    
+    /**
+     * Removes an SVGObject of matching id nested from the current SVGObject.
+     * Uses depths first search.
+     * @param id
+     */
+    public final void removeChild(final String id) {
+        this.svgDataReference.removeChild(id);
+    }
+    
+    private List<SVGObject> getAllChildren() {
+        final ArrayList<SVGObject> children = new ArrayList<>();
+        this.svgDataReference.getAllChildren().forEach(child -> children.add(new SVGObject(child)));
+        return children;
+    }   
+    
+    /**
+     * Sets the content value of an SVGObjects.
+     * Content is considered plain text that lies within a SVG element 
+     * such as is the case with text elements.
+     * @param content
+     */
+    public void setContent(final String content) {
+        this.svgDataReference.setContent(SVGParser.sanitisePlanText(content));
+    }
+    
+    /**
+     * Sets the id value of the SVGObject.
+     * This value must be unique, or may produce undesirable results.
+     * id values should be descriptive.
+     * Uniqueness is not currently enforced.
+     * @param id 
+     */
+    public void setID(final int id) {
+        this.setAttribute(SVGAttributeConstants.ID, id);
+    }
+    
+    /**
+     * Sets the id value of the SVGObject.
+     * This value must be unique, or may produce undesirable results.
+     * id values should be descriptive.
+     * Uniqueness is not currently enforced.
+     * @param id 
+     */
+    public void setID(final String id) {
+        this.setAttribute(SVGAttributeConstants.ID, id);
+    }
+    
+    /**
+     * Gets the id value of the SVGObject.
+     * Will be useful when parent element id's prefix their child element id's to reflect their position in the SVGData structure.
+     * @return idString
+     */
+    public final String getID() {
+        return svgDataReference.getAttributeValue(SVGAttributeConstants.ID.getName());
+    }
+    
+    /**
+     * Adds a String attribute to the current SVGData. 
+     * @param attributeKey 
+     * @param attributeValue 
+     */
+    private void setAttribute(final SVGAttributeConstants attributeKey, final String attributeValue) {
+        svgDataReference.setAttribute(attributeKey, attributeValue);
+    }
+    
+    /**
+     * Adds a float attribute to the current SVGData. 
+     * @param attributeKey 
+     * @param attributeValue 
+     */
+    private void setAttribute(final SVGAttributeConstants attributeKey, final float attributeValue) {
+        svgDataReference.setAttribute(attributeKey, String.format("%.2f", attributeValue));
+    }
+    
+    /**
+     * Adds an int attribute to the current SVGData. 
+     * @param attributeKey 
+     * @param attributeValue 
+     */
+    private void setAttribute(final SVGAttributeConstants attributeKey, final int attributeValue) {
+        svgDataReference.setAttribute(attributeKey, String.format("%s", attributeValue));
```

Comment:
instead of using a string formatter here, could use `String.valueOf(attributeValue)`